tests/common/beamlines/test_beamline_utils.py

```diff
@@ -9,14 +13,22 @@
 from ophyd_async.core import Device as OphydV2Device
 from ophyd_async.core import StandardReadable
 
+import dodal.common.beamlines.beamline_utils as beamline_utils
 from dodal.common.beamlines import beamline_utils
+from dodal.common.beamlines.beamline_utils import (
+    clear_path_provider,
+    get_path_provider,
+    set_path_provider,
+)
 from dodal.devices.eiger import EigerDetector
 from dodal.devices.focusing_mirror import FocusingMirror
 from dodal.devices.motors import XYZStage
 from dodal.devices.smargon import Smargon
 from dodal.log import LOGGER
 from dodal.utils import DeviceInitializationController
 
+UUID_REGEX = re.compile(r"^[\da-f]{8}-([\da-f]{4}-){3}[\da-f]{12}$", re.IGNORECASE)
```

Comment:
Can we just try creating a uuid from the string? The UUIDFileNameProvider takes a callable that returns a string, for right now that's `uuid.uuid4` but I wouldn't put it past someone to change the behaviour in unexpected ways.